### PR TITLE
feat: implement `hyperlane warp deploy`

### PR DESCRIPTION
**Pull request by ltyu**
### Description
- Update `hyperlane warp deploy` command

### Drive-by changes
- Move Core deploy to core.ts
- Update Module return args to match TokenType

### Related issues
- Fixes #[3541](https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3541)

### Backward compatibility
yes, but old command no longer applies

### Testing
Manual

### CODE REVIEW

File: typescript/sdk/src/token/EvmERC20WarpModule.ts [hunk 1]
```diff
@@ -84,7 +76,7 @@ export class EvmERC20WarpModule extends HyperlaneModule<
 
     return new EvmERC20WarpModule(multiProvider, {
       addresses: {
-        deployedTokenRoute: deployedContracts[config.type].address,
+        [config.type]: deployedContracts[config.type].address,
```
Comment (cmcewen) [inline typescript/sdk/src/token/EvmERC20WarpModule.ts:79]: does this make type-ing harder to reason about?

File: typescript/cli/src/deploy/warp.ts [hunk 2]
```diff
@@ -110,20 +112,30 @@ async function executeDeploy(params: DeployParams) {
...
+        multiProvider,
+      });
 
+      return evmERC20WarpModule.serialize();
+    }),
+  );
   logGreen('✅ Hyp token deployments complete');
 
-  const warpCoreConfig = await getWarpCoreConfig(params, deployedContracts);
+  const deployedContractMap = attachContractsMap(
```
Comment (cmcewen) [inline typescript/cli/src/deploy/warp.ts:133]: can you do this inside the map above? or this function expects a map vs just attaching as single one at a time

Comment (ltyu) [reply typescript/cli/src/deploy/warp.ts]: Yup, i will keep that in mind when i re-implement this in another PR

File: typescript/cli/src/commands/core.ts [hunk 3]
```diff
@@ -34,6 +43,48 @@ export const coreCommand: CommandModule = {
...
+  chain: string;
+  config: string;
+  dryRun: string;
+  fromAddress: string;
+}> = {
+  command: 'deploy',
+  describe: 'Deploy Hyperlane contracts',
+  builder: {
+    chain: chainCommandOption,
+    config: outputFileCommandOption(
```
Comment (yorhodes) [inline typescript/cli/src/commands/core.ts:62]: should this be input ?

**Review by yorhodes — APPROVED** [review]
lets revert the address shape change but otherwise lgtm

File: typescript/cli/src/commands/core.ts (hunk 3 above)
Comment (ltyu) [reply typescript/cli/src/commands/core.ts:62]: yes

File: typescript/sdk/src/token/EvmERC20WarpModule.ts (hunk 1 above)
Comment (ltyu) [reply typescript/sdk/src/token/EvmERC20WarpModule.ts:79]: reverted

File: typescript/cli/src/commands/core.ts (hunk 3 above)
Comment (ltyu) [reply typescript/cli/src/commands/core.ts:62]: Will address in a following PR. Requires refactoring of the input function, but also merging this for bug bash

**Pull request merged**